### PR TITLE
Add engines to package.json during firebase init functions

### PR DESCRIPTION
**Pull request by joehan**
### Description
Adding an explicit engines field to package.json created during init.
Addresses internal bug 119619092

### CODE REVIEW

**Comment by joehan** [comment]
@mbleigh Wanted to run this by you first - are you ok with switching these to default to Node 8?

File: templates/init/functions/javascript/eslint.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "parserOptions": {
     // Required for certain syntax usages
-    "ecmaVersion": 6
+    "ecmaVersion": 8
```
Comment (mbleigh) [inline templates/init/functions/javascript/eslint.json]: Is this actually right? I don't think `ecmaVersion` corresponds to Node.js version. There is ES6, but there is no ES8

Comment (bkendall) [reply templates/init/functions/javascript/eslint.json]: Huh, [it is](https://eslint.org/docs/user-guide/configuring):

> ecmaVersion - set to 3, 5 (default), 6, 7, 8, 9, or 10 to specify the version of ECMAScript syntax you want to use. You can also set to 2015 (same as 6), 2016 (same as 7), 2017 (same as 8), 2018 (same as 9), or 2019 (same as 10) to use the year-based naming.

That said, personally I'd stick with `es2015` (or `6`) as that's supported by Node `8` just fine. (also, sticking with `esYYYY` to match `tsconfig.json`)

Comment (thechenky) [reply templates/init/functions/javascript/eslint.json]: Let's change this to es2017 since we're upgrading everything to 8.

Comment (mbleigh) [reply templates/init/functions/javascript/eslint.json]: +1 to es2017 since that includes async/await

**Comment by thechenky** [comment]
@joehan can you make the requested changes? We need to release this together with the Node6/8/10 upgrades.

**Comment by thechenky** [comment]
This should be ready to merge now - please sync with master and merge.

**Comment by thechenky** [comment]
We are releasing in an hour, please merge.

**Pull request merged**